flink-streaming-java/src/main/java/org/apache/flink/streaming/api/checkpoint/ListCheckpointed.java

```diff
@@ -41,6 +41,7 @@
 	 * @param checkpointId The ID of the checkpoint.
 	 * @param timestamp Timestamp of the checkpoint.
 	 * @return The operator state in a list of redistributable, atomic sub-states.
+	 *         Should not return null, but empty list instead.
```

Comment:
That is why I wrote "should". It is possible, but not encouraged because i) `null` is evil ;-) and ii) user will also receive empty list on restore.